warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -889,7 +890,12 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
         
         TraceStopwatch stopwatch = null;
         
-        if (!disableExpandIndexFunction) {
+        // Parse out some metadata about the query plan to make informed decisions as the query tree is processed.
+        stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Build QueryPlanMetadata");
+        QueryPlanMetadata planMetadata = QueryPlanMetadataVisitor.getQueryPlanMetadata(queryTree);
+        stopwatch.stop();
+        
+        if (planMetadata.hasFunctionNodes() && !disableExpandIndexFunction) {
```

Comment:
Does this actually save us any time if we simply need to re-gather the plan metadata once anything has changed?